### PR TITLE
pkg/acpi: lift loop condition and remove Sprintf

### PR DESCRIPTION
**Pull request by binjip978**

### CODE REVIEW

File: pkg/acpi/raw.go [hunk 1]
```diff
@@ -102,7 +99,7 @@ func (r *Raw) TableData() []byte {
 
 // Sig returns the table signature.
 func (r *Raw) Sig() string {
-	return fmt.Sprintf("%s", r.data[:4])
+	return string(r.data[:4])
```
Comment (rminnich) [inline pkg/acpi/raw.go:102]: qq, are we sure at this point that data is long enough?

**Review by rminnich — COMMENTED** [review]
see my question, I've started doing silly things like this go make sure it will work, but there has to be a better way to do this
	var b [4]byte
	var data []byte
	bytes.NewBuffer(data).Read(b[:])
	fmt.Printf("%q\n", string(b[:]))
kinda gross, but safe. is there not some package that does this? I've looked in some of the generic packages but not found it yet. But .. it ought to exist.


still, we've had this package for years, but still ...

File: pkg/acpi/raw.go (hunk 1 above)
Comment (binjip978) [reply pkg/acpi/raw.go:102]: No we don't, never been and this code can panic, but this transformation is save.

**Comment by binjip978** [comment]
> var b [4]byte
> var data []byte
> bytes.NewBuffer(data).Read(b[:])
> fmt.Printf("%q\n", string(b[:]))

What the difference with this snippet:

```go
data := make([]byte, 4)
fmt.Printf("%q\n", string(data[:]))
```

We will got "\x00\x00\x00\x00" in both cases?

**Comment by rminnich** [comment]
The only difference: try it with 
var b [0]byte
var data []byte
bytes.NewBuffer(data).Read(b[:])
fmt.Printf("%q\n", string(b[:]))

**Review by rminnich — APPROVED** [review]
it's a google tradition in a PR to ask you to fix stuff somebody else wrote wrong :-)

**Pull request merged**